library/core/src/iter/traits/collect.rs

```diff
@@ -146,6 +146,7 @@ pub trait FromIterator<A>: Sized {
     /// assert_eq!(v, vec![5, 5, 5, 5, 5]);
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_diagnostic_item = "from_iterator_method"]
```

Comment:
```suggestion
    #[rustc_diagnostic_item = "from_iter_method"]
```

Reply:
Also it's not even a method, so maybe even just
```suggestion
    #[rustc_diagnostic_item = "from_iter"]
```
or
```suggestion
    #[rustc_diagnostic_item = "from_iter_function"]
```
or
```suggestion
    #[rustc_diagnostic_item = "from_iterator_from_iter"]
```

Reply:
For all the other single function traits they can be interpreted as `trait_name_method` or `method_name_method`. Either one is a reasonable way to name them. It is definitely not a method though.

Reply:
@Jarcho could you change this to `*_fn` the same way you did with other assoc functions? I'd suggest `from_iter_fn`.